src/apps/media/src/app/components/InsightsTable.tsx

```diff
@@ -0,0 +1,252 @@
+import { useState, useMemo, useRef } from "react";
+import {
+  Box,
+  Typography,
+  Card,
+  Chip,
+  CardMedia,
+  IconButton,
+} from "@mui/material";
+import { useSelector } from "react-redux";
+import {
+  useGetUsageQuery,
+  useGetRequestsQuery,
+} from "../../../../../shell/services/metrics";
+import { DataGridPro, GridValueGetterParams } from "@mui/x-data-grid-pro";
+import { File, Bin } from "../../../../../shell/services/types";
+import fileBroken from "../../../../../../public/images/fileBroken.jpg";
+import { useHistory, useLocation } from "react-router-dom";
+
+import {
+  fileExtension,
+  fileTypeToColor,
+  getExtensions,
+  getDateFilterFn,
+} from "../utils/fileUtils";
+import { AppState } from "../../../../../shell/store/types";
+import {
+  useGetAllBinFilesQuery,
+  useGetBinsQuery,
+} from "../../../../../shell/services/mediaManager";
+import LinkRoundedIcon from "@mui/icons-material/LinkRounded";
+import CheckIcon from "@mui/icons-material/Check";
+
+export const InsightsTable = () => {
+  // Thumbnail prerequisites
+  const imageEl = useRef<HTMLImageElement>();
+  const [imageOrientation, setImageOrientation] = useState<string>("");
```

Comment:
I'm saving it in case @zcolah wants to apply the same vertical and horizontal thumbnail display. But I think this wouldn't be necessary for now. Will remove it.